Add Tests and Demos about "Sensitive Data Stored Unencrypted via Room" (#3541)
OWASP/mastg: tests-beta/android/MASVS-STORAGE/MASTG-TEST-0x01.md

---
platform: android

> title: Runtime Verification of Sensitive Data Stored Unencrypted in Android Room DB

Reviewer: ```suggestion
title: Runtime Storage of Unencrypted Data in Room Databases
```

id: MASTG-TEST-0x01
type: [dynamic, filesystem]
weakness: MASWE-0006

> best-practices: [MASTG-BEST-0x25]

Reviewer: BEST-0x25 is a new BEST file? Maybe you can make a BEST file for Android that recommends Storing Data Encrypted in App Sandbox Directory, like the one that exists for iOS right now (https://mas.owasp.org/MASTG/best-practices/MASTG-BEST-0024/). 

You can attach this BEST file here, or in the other PRs, and expand it with examples of each of the ways to store data (Room, SQLite, DataStore) - If you are not doing this now, please create a PR to keep track of the needs of creating a file like this for Android.

Author: Best 0x25 is part of PR #3534

profiles: [L2]
status: new
---

## Overview

> This test checks at runtime whether sensitive data — tokens, secrets, or PII — is stored in Room databases without encryption. The goal is to ensure that sensitive information is not persisted in plaintext within the app's private storage.

Reviewer: ```suggestion
This test complements @MASTG-TEST-0x02. It checks at runtime whether sensitive data (like tokens, secrets, or PII) is stored in Room databases without encryption.

If the app stores sensitive data in the Room databases, such as tokens, credentials, or PII, without integrating an encryption layer like SQLCipher (@MASTG-KNOW-0038), anyone who gains access to the app's sandbox (via physical access or backup extraction) can extract the plaintext data.

The goal is to ensure that sensitive information is not persisted in plaintext in the Room databases within the app's private storage.
```

## Steps

1. Exercise all the functionalities of the app that process or store sensitive data.

2. Access the app's private storage (@MASTG-TECH-0008) and locate Room database files:
   - `/data/data/<package_name>/databases/<database_name>`
   - `/data/data/<package_name>/databases/<database_name>-wal`
   - `/data/data/<package_name>/databases/<database_name>-shm`

3. Extract the database files of the app to the host machine (@MASTG-TECH-0002).

4. Inspect database contents using a SQLite client or dynamic analysis tool to confirm whether sensitive data is stored in plaintext.

## Observation

> 

Reviewer: ```suggestion
The output should contain:
```
